pkg/operator/staticpod/controller/installer/installer_controller.go

```diff
@@ -214,15 +214,21 @@ func (c *InstallerController) manageInstallationPods(operatorSpec *operatorv1.Op
 			// it's an extra write/read, but it makes the state debuggable from outside this process
 			if !equality.Semantic.DeepEqual(newCurrNodeState, currNodeState) {
 				glog.Infof("%q moving to %v", currNodeState.NodeName, spew.Sdump(*newCurrNodeState))
-				operatorStatus.NodeStatuses[i] = *newCurrNodeState
-				updateError := c.updateStatus(resourceVersion, originalOperatorStatus, operatorStatus)
-				if updateError == nil {
-					if currNodeState.CurrentRevision != newCurrNodeState.CurrentRevision {
-						c.eventRecorder.Eventf("NodeCurrentRevisionChanged", "Updated node %q from revision %d to %d", currNodeState.NodeName,
-							currNodeState.CurrentRevision, newCurrNodeState.CurrentRevision)
+				if updated, updateError := common.UpdateStatus(c.operatorConfigClient, func(operatorStatus *operatorv1.StaticPodOperatorStatus) error {
+					for i := range operatorStatus.NodeStatuses {
```

Comment:
you inlined this just to annoy me, didn't you?

Reply:
have you ever done https://en.wikipedia.org/wiki/Continuation-passing_style ? Should bring you a nice book about that for xmas, with exercises.